### ISSUE TITLE
postgres: insert_many should use UNNEST and ARRAYs

### ISSUE THREAD

**Issue by gitmalong** [SeaQL/sea-orm#1922]
See my comment here https://github.com/SeaQL/sea-orm/discussions/710#discussioncomment-7319840

**Comment by tyt2y3** [comment]
I imagine we can have a bulk insert API that bypasses ActiveModel. Since UNNEST ARRAY is column-oriented.

Something like:

```rust
MyEntity::bulk_insert([
    (my_entity::Column::Id, &[1,2,3,4]),
    (my_entity::Column::Name, &["a","b","c","d"]),
])
```

That'd build:

```sql
INSERT INTO my_entity (id, name) VALUES UNNEST ARRAY($1, $2);
# where $1 is i32[], $2 is test[]
```

**Comment by gitmalong** [comment]
From end users perspective I would still like to pass over my whole `Vec<MyStruct>` and let SeaORM take care of the "field to vec split" instead of setting each column on my own because it is easy to forget something etc. But I don't know about the internals and if we can first start with your suggested `bulk_insert` and then build something more ergonomic on top of it.

**Comment by tyt2y3** [comment]
Yes, that sounds great. We can may be add a `fn into_bulk_insert<E: EntityTrait>(items: Vec<E::ActiveModel>) -> BulkInsert<E>`

**Comment by Expurple** [comment]
> I imagine we can have a bulk insert API that bypasses ActiveModel.

> Something like:

> ```rust
> MyEntity::bulk_insert([
>     (my_entity::Column::Id, &[1,2,3,4]),
>     (my_entity::Column::Name, &["a","b","c","d"]),
> ])
> ```

> That'd build:

> ```sql
> INSERT INTO my_entity (id, name) VALUES UNNEST ARRAY($1, $2);
> # where $1 is i32[1,2,3,4], $2 is text["a","b","c","d"]
> ```

If you're OK with bypassing `ActiveModel` and spelling out the columns like that, then you don't need any new `sea_orm` APIs.

This can be achieved with raw SQL today:

```rust
let values = [
    Value::from(vec![1, 2, 3, 4]),
    Value::from(Vec::from(["a", "b", "c", "d"].map(|s| s.to_string()))),
];
let statement = Statement::from_sql_and_values(
    DatabaseBackend::Postgres,
    "INSERT INTO my_entity (id, name) VALUES UNNEST ARRAY($1, $2);",
    values,
);
tx.execute(statement).await?;
```

And if you need a dynamic query or want to use Rust identifiers, you could render the SQL string from a [`sea_query::InsertStatement`](https://docs.rs/sea-query/0.32.6/sea_query/query/struct.InsertStatement.html) instead of hardcoding it:

```rust
let values = [
    Value::from(vec![1, 2, 3, 4]),
    Value::from(Vec::from(["a", "b", "c", "d"].map(|s| s.to_string()))),
];
let query = Query::insert()
    .into_table(MyEntity::Table)
    .columns([MyEntity::Id, MyEntity::Name])
    .values_panic([Expr::cust_with_values("UNNEST ARRAY($1, $2)", values).into_simple_expr()])
    .to_owned();
let (sql, Values(values)) = query.build_any(&PostgresQueryBuilder);
let statement = Statement::from_sql_and_values(DatabaseBackend::Postgres, sql, values);
tx.execute(statement).await?;
```

This code compiles, but I haven't actually run it to check whether this "`cust_with_values` -> `values_panic` -> `build_any`" pipeline is smart enough to preserve and return the same `values` that we've put in.

This example hardcodes the number of columns in the raw `"UNNEST ARRAY($1, $2)"`, but it can probably be changed to something dymanic like

```rust
Func::cust("UNNEST").arg(Func::cust("ARRAY").args(values.map(|v| v.into())))
```

---

I agree that it would be better to have an ergonomic API that accepts `ActiveModel`s and does all of this under the hood.

I think, we don't need a new API for that either. As described in #710, the existing `insert_many` should do this by default to avoid suprising parameter limit errors (which are usually not hit during development and are discovered in production)
